### PR TITLE
Add Tag Support to Resource Group Resource

### LINKED ISSUE #7109: Support resource tags for aws_resourcegroups_group resource
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

We recently merged in the initial `aws_resourcegroups_group` resource for managing the group, but it is currently missing resource tag support. It can be implemented similar to other AWS service resources that support tags via the `Tag`, `UnTag`, and `GetTags` API calls. Please note: this feature request is for tagging the group itself, not to support tagging other resources via the group.

### New or Affected Resource(s)

* `aws_resourcegroups_group`

### Potential Terraform Configuration

```hcl
resource "aws_resourcegroups_group" "example" {
  # ... other configuration ...
  tags = {
    Key1 = "Value1"
  }
}
```

### References

* #2972 
* #6217

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

Closes #7109

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
- resource/resource_aws_resourcegroups_group: Add `tags` argument.
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSResourceGroup_'

=== RUN   TestAccAWSResourceGroup_basic
=== PAUSE TestAccAWSResourceGroup_basic
=== CONT  TestAccAWSResourceGroup_basic
--- PASS: TestAccAWSResourceGroup_basic (85.00s)
=== RUN   TestAccAWSResourceGroup_tags
=== PAUSE TestAccAWSResourceGroup_tags
=== CONT  TestAccAWSResourceGroup_tags
--- PASS: TestAccAWSResourceGroup_tags (114.05s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	139.697s

...
```

I saw that there is pending PR for this #7811 which i only realised only after i was done with this.
this refactors the tests as well as uses the keyvaluetags lib

### CODE REVIEW

**Comment by bflad** [comment]
Hi @DrFaust92 👋 Thanks for submitting this one as well.

Looks like there's a bit of a backlog for adding tagging for this resource. The maintainers prefer working with contributions in their submission order to be fair to contributors. In this case, that means we should try working with the original author (even before #7811 there is #7774) to either help review their contribution, asking them to close their contribution, or if they are unresponsive to submit a fix on top of their commits unless they close their pull request so they receive contributing credit.

Before we even go through that though, it might be best to get the `keyvaluetags` updates submitted separately since they appear to be in merge conflict already in this pull request. To prevent this situation as well as help human readers, we would encourage alphabetizing additions to `switch` statements and `import` declarations.

Please reach out with any questions.

**Comment by DrFaust92** [comment]
@bflad, ill open up a separate PR for the `keyvaluetags` change for this resource and reach out to other contributors for this. 
ill keep this open for now and close if the change will be merged to another PR
thanks!

**Comment by DrFaust92** [comment]
fixed merge conflicts

```
$ make testacc TESTARGS='-run=TestAccAWSResourceGroup_'

=== RUN   TestAccAWSResourceGroup_basic
=== PAUSE TestAccAWSResourceGroup_basic
=== CONT  TestAccAWSResourceGroup_basic
--- PASS: TestAccAWSResourceGroup_basic (67.41s)
=== RUN   TestAccAWSResourceGroup_tags
=== PAUSE TestAccAWSResourceGroup_tags
=== CONT  TestAccAWSResourceGroup_tags
--- PASS: TestAccAWSResourceGroup_tags (94.25s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	181.002s

```

**Comment by rsareth** [comment]
Please, merge it!

File: aws/resource_aws_resourcegroups_group_test.go [hunk 1]
```diff
@@ -75,7 +63,52 @@ func TestAccAWSResourceGroup_basic(t *testing.T) {
 	})
 }
 
-func testAccCheckAWSResourceGroupExists(n string) resource.TestCheckFunc {
+func TestAccAWSResourceGroup_tags(t *testing.T) {
+	var v resourcegroups.Group
+	resourceName := "aws_resourcegroups_group.test"
+	n := fmt.Sprintf("test-group-%d", acctest.RandInt())
+	desc1 := "Hello World"
```
Comment (bflad) [inline aws/resource_aws_resourcegroups_group_test.go]: Nit: This acceptance testing does not need to verify anything with this attribute, so it should omitted or hardcoded in the test configurations.

File: aws/resource_aws_resourcegroups_group_test.go [hunk 2]
```diff
@@ -138,3 +193,42 @@ JSON
...
+
+func testAccAWSResourceGroupConfigTags1(rName, desc, query, tag1Key, tag1Value string) string {
+	return fmt.Sprintf(`
+resource "aws_resourcegroups_group" "test" {
+  name        = "%s"
+  description = "%s"
+
+  resource_query {
+    query = <<JSON
+%s
```
Comment (bflad) [inline aws/resource_aws_resourcegroups_group_test.go]: Nit: We prefer to hardcode values like these into the test configurations so they can more easily be copy-pasted for manual Terraform CLI usage.

**Pull request merged**